src/features/game/events/landExpansion/buySeasonalItem.test.ts

```diff
@@ -119,4 +119,56 @@ describe("buySeasonalItem", () => {
 
     expect(result.wardrobe["Red Farmer Shirt"]).toEqual(1);
   });
+
+  it("throws an error if key already bought today", () => {
+    expect(() =>
+      buySeasonalItem({
+        state: {
+          ...mockState,
+          inventory: {
+            "Treasure Key": new Decimal(0),
+            Wood: new Decimal(100),
+          },
+          pumpkinPlaza: {
+            keysBought: {
+              factionShop: {},
+              treasureShop: {},
+              megastore: {
+                "Treasure Key": { boughtAt: new Date("2024-08-09").getTime() },
+              },
+            },
+          },
+        },
+        action: {
+          type: "seasonalItem.bought",
+          name: "Treasure Key",
+          tier: "basic",
+        },
+        createdAt: new Date("2024-08-09").getTime(),
+      }),
+    ).toThrow("Already bought today");
```

Comment:
Maybe the user-facing shopping UX needs to have an indicator (and/or text) that some items have a once-per-day limit?

Reply:
I think this related to the keys logic - there is currently a "Daily limit - 1/1" label that helps out 😄